### ISSUE TITLE
From error state to other state on a Conversational bot

### ISSUE THREAD

**Issue by paulthemagno** [python-telegram-bot/python-telegram-bot#2277]
I have a ConversationHandler with an `error` function like this:
```python
def error(update, context):
    logger.print_log("got the error:" + str(context.error))
```
My bot is like this:

```python
updater = Updater(token,use_context=True)
dp = updater.dispatcher
conv_handler = ConversationHandler(
        entry_points=[CommandHandler('start', start)],
        states={
            
            STATE1:    [...],
            STATE2:  [...],
            ...
        },
        fallbacks=[CommandHandler('cancel', cancel)]
    )
dp.add_handler(conv_handler)
dp.add_error_handler(error)
updater.start_polling()
updater.idle()
```

I'm trying to treat `error` function like all the other states, but it seems not working. After the error the program remains in the same state where the error has occurred, but I want to go always to STATE2. How can I do?

**Comment by Bibo-Joshi** [comment]
the error handlers are completely independent from any registered conversation handlers If you want your conversation to change states depending on raised exceptions, you should handle the exceptions within the conversation.

**Comment by Poolitzer** [comment]
I like the idea of having different error handlers based on parameters though. Either with an error state in the conv handler, or maybe filters at the error handler?

**Comment by Bibo-Joshi** [comment]
Mh, I had a simliar [discussion](https://github.com/python-telegram-bot/python-telegram-bot/pull/2051#discussion_r491682301) with Noam. IMO a handler-like setup for error handlers is over the top. If you need special casing about where you error originated from, you might as well do a `try-except` where you acutally need it.

However I do see that `ConversationHandler` is somewhat of a special case. But we would probably have to somehow make the error handlers in a `error_handlers` list aware of which state the error occured in … The most precise way would be to add error handlers for each state but then we're bag to just `try-except` in the callback.
TL;DR: Not too sure about it. But probably worth to at least discuss

**Comment by paulthemagno** [comment]
@Bibo-Joshi Yes, I did so. I add a 
```python
try:
   ...
except Exception as e:
   return handle_error(error=e, update=update)
```
in every state and the function
```python
def handle_error(error, update):
    tb = traceback.format_exc()
    print("got the error:" + str(error)+"\nMore info:" + str(tb) , error=True)
    print(tb)
    message = "Ops! Error😞"
    keyboard = [[InlineKeyboardButton("Close❌", callback_data='cancel')],
                [InlineKeyboardButton("Menù🏠↩️", callback_data='menu')]]
    reply_markup = InlineKeyboardMarkup(keyboard,
                                    one_time_keyboard=True,
                                    resize_keyboard=True)
    if update.message != None:
        update.message.reply_text(message, reply_markup=reply_markup,parse_mode=ParseMode.HTML)
    else:
        updater.bot.answerCallbackQuery(update.callback_query.id)
        update.callback_query.edit_message_text(message, reply_markup=reply_markup,parse_mode=ParseMode.HTML)

    return HANDLE_ERROR
```

So each error is handled by this function and it works. A bit redundant, but working

**Comment by Poolitzer** [comment]
> A bit redundant, but working

I guess you could have used a decorator if you want to minimize redundancy :)

**Comment by Bibo-Joshi** [comment]
After internal discussion, we don't see this feature being implemented, at least not in the forseable future. If someone finds this comment and really wants this feature: Feel free to comment or open a new issue that references this one.
